core/templates/dev/head/domain/summary/ExplorationSummaryBackendApiService.js

```diff
@@ -19,9 +19,25 @@
 
 oppia.factory('ExplorationSummaryBackendApiService', [
   '$http', '$q', 'EXPLORATION_SUMMARY_DATA_URL_TEMPLATE',
-  function($http, $q, EXPLORATION_SUMMARY_DATA_URL_TEMPLATE) {
-    var _fetchExpSummaries = function(explorationIds,
-      includePrivateExplorations, successCallback, errorCallback) {
+  'validatorsService', 'alertsService',
+  function(
+      $http, $q, EXPLORATION_SUMMARY_DATA_URL_TEMPLATE,
+      validatorsService, alertsService) {
+    var _fetchExpSummaries = function(
+        explorationIds, includePrivateExplorations, successCallback,
+        errorCallback) {
+      if (!explorationIds.every(validatorsService.isValidExplorationId)) {
+        alertsService.addWarning('Please enter a valid exploration ID.');
```

Comment:
Actually, in this case, app.js is called with second argument `undefined`, so the warning there doesn't trigger. I coded the app.js part for parity with the other similar functions in the validator service.